tools/roundtrip_test.sh

```diff
@@ -0,0 +1,89 @@
+#!/bin/bash
+# Copyright (c) the JPEG XL Project Authors. All rights reserved.
+#
+# Use of this source code is governed by a BSD-style
+# license that can be found in the LICENSE file.
+
+# End-to-end roundtrip tests for cjxl and djxl tools.
+
+MYDIR=$(dirname $(realpath "$0"))
+JPEGXL_TEST_DATA_PATH="${MYDIR}/../testdata"
+
+set -eux
+
+# Temporary files cleanup hooks.
+CLEANUP_FILES=()
+cleanup() {
+  if [[ ${#CLEANUP_FILES[@]} -ne 0 ]]; then
+    rm -rf "${CLEANUP_FILES[@]}"
+  fi
+}
+trap 'retcode=$?; { set +x; } 2>/dev/null; cleanup' INT TERM EXIT
+
+roundtrip_test() {
+  local infn="${JPEGXL_TEST_DATA_PATH}/$1"
+  local encargs="$2"
+  local maxdist="$3"
+  
+  local encoder="${build_dir}/tools/cjxl"
+  local decoder="${build_dir}/tools/djxl_ng"
+  local comparator="${build_dir}/tools/ssimulacra_main"
```

Comment:
In some build, like making the deb packages, `/tools/ssimulacra_main` is not build, see https://github.com/libjxl/libjxl/runs/6902801608?check_suite_focus=true